Doc WITH ORDINALITY (#1345)
risingwavelabs/risingwave-docs: docs/sql/query-syntax/query-syntax-with-ordinality-clause.md

---
id: query-syntax-with-ordinality-clause
slug: /query-syntax-with-ordinality-clause
title: WITH ORDINALITY clause
---

> The `WITH ORDINALITY` clause can only be used with set-returning functions at the end of a `SELECT` statement. An additional integer column will be added to the table, which numbers the rows returned by the function, starting from 1. By default, the generated column is named `ordinality`. 

Reviewer: ```suggestion
The `WITH ORDINALITY` clause can be used with set-returning functions in the `FROM` clause of a query. An additional integer column will be added to the table, which numbers the rows returned by the function, starting from 1. By default, the generated column is named `ordinality`. 
```

- Not at the end of a `SELECT` statement, but in the `FROM` clause. e.g., `FROM func() WITH ORDINALITY JOIN table` is ok.
- Strictly speaking, it can also be used with scalar functions, but that isn't very useful though.

> See [Set functions](/sql/function-operators/sql-function-set-functions.md) for a list of supported set-returning functions.

Reviewer: nit: why the page's name is "Set functions" while the functions are "set-returning functions"? It feels inconsistent. 🤣

Author: Thanks for mentioning that!
